src/test/java/org/jenkinsci/plugins/gitclient/GitClientTest.java

```diff
@@ -597,6 +600,41 @@ public void testAddRemoteUrl() throws Exception {
         assertEquals(upstreamRepoURL, gitClient.getRemoteUrl("upstream"));
     }
 
+    @Test (expected = GitException.class)
+    public void testAutocreateFailsOnMultipleMatchingOrigins() throws Exception {
+        File repoRootTemp = tempFolder.newFolder();
+        GitClient gitClientTemp = Git.with(TaskListener.NULL, new EnvVars()).in(repoRootTemp).using(gitImplName).getClient();
+        gitClientTemp.init();
+        FilePath gitClientFilePath = gitClientTemp.getWorkTree();
+        FilePath gitClientTempFile = gitClientFilePath.createTextTempFile("aPre", ".txt", "file contents");
+        gitClientTemp.add(".");
+        gitClientTemp.commit("Added " + gitClientTempFile.toURI().toString());
+        gitClient.clone_().url("file://" + repoRootTemp.getPath()).execute();
+        final URIish remote = new URIish(Constants.DEFAULT_REMOTE_NAME);
+
+        try ( // add second remote
+              FileRepository repo = new FileRepository(new File(repoRoot, ".git"))) {
+            StoredConfig config = repo.getConfig();
+            config.setString("remote", "upstream", "url", "file://" + repoRootTemp.getPath());
+            config.setString("remote", "upstream", "fetch", "+refs/heads/*:refs/remotes/upstream/*");
+            config.save();
+        }
+
+        // fill both remote branches
+        List<RefSpec> refspecs = Collections.singletonList(new RefSpec(
+                "refs/heads/*:refs/remotes/origin/*"));
+        gitClient.fetch_().from(remote, refspecs).execute();
+        refspecs = Collections.singletonList(new RefSpec(
+                "refs/heads/*:refs/remotes/upstream/*"));
+        gitClient.fetch_().from(remote, refspecs).execute();
+
+        // checkout will fail
+        gitClient.checkout().ref(Constants.MASTER).execute();
```

Comment:
I assume this is the statement that throws the assertion.  If so, then the next two statements won't be executed.  In that case, it may be better to use the try / catch that you had before, but then insert an assertion into the catch block which checks some specific value of the GitException.
```suggestion
        gitClient.checkout().ref(Constants.MASTER).execute();
```